nemo/collections/tts/parts/utils/helpers.py

```diff
@@ -825,3 +826,14 @@ def sample_tts_input(
             0, export_config["num_speakers"], (max_batch,), device=device, dtype=torch.int64
         )
     return inputs
+
+
+@deprecated(
+    explanation="But it will not be removed until a further notice. G2P object root directory "
+    "`nemo_text_processing.g2p` has been replaced with `nemo.collections.tts.g2p`. "
```

Comment:
Should we commit to a particular release to remove this? Maybe within 1-2 releases?

Reply:
I think 1.18 was the release that we are trying to remove this replacement for `nemo_text_processing.g2p`, which means all old models before 1.18 will raise errors for the import path. We have no plan now to re-train all old models. But we can modify the state_dict for all old models if we plan to remove such deprecation warnings and back-compatibility replacement.

Reply:
So for now, it is good to keep the warning and replacement op until re-train or update state_dict is done.

Reply:
I don't see why retraining all of our models would be a priority (unless we have several other deprecations we also want to resolve).

But if we do not commit to a timeline, then it would be clearer to only print the deprecation warning when the checkpoint is using the old class. Will this annotation result in all model restorations printing the message?

Reply:
good point. This warning should be only triggered when model class uses `nemo_text_processing.g2p`.